### PR TITLE
http2: Extract conn tuple from http2_stream_t

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Removes `conn_tuple_t` field from `http2_stream_t` struct.
Similar to https://github.com/DataDog/datadog-agent/pull/20942

### Motivation

We can remove the instance of conn_tuple_t from http2_stream_t, and by that minimize the size of http2_in_flight map. The conn_tuple_t field in http2_stream_t is redundant as it exists in the key of the map. Alongside we removing the duplication, we introduce a new struct called http2_event_t which composes both conn_tuple_t and http2_stream_t struct, and send http2_event_t to the user mode. By that, we reduce the size of http2_in_flight without affecting the events sent to the usermode.

### Additional Notes
Before:
```
57574: hash  name http2_in_flight  flags 0x0
	key 56B  value 232B  max_entries 65536  memlock 18874368B
	btf_id 121637
```
After:
```
57465: hash  name http2_in_flight  flags 0x0
	key 56B  value 184B  max_entries 65536  memlock 15728640B
	btf_id 121474
```

Reduces 3MB (16%) from the map size (and the agent's WSS) for 65536 entries. The change is linear with the number of entries.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http2/decoding.h [hunk 1]
```diff
@@ -280,10 +280,16 @@ static __always_inline void handle_end_of_stream(http2_stream_t *current_stream,
 
     // response end of stream;
     current_stream->response_last_seen = bpf_ktime_get_ns();
-    current_stream->tup = http2_stream_key_template->tup;
 
-    // enqueue
-    http2_batch_enqueue(current_stream);
+    const u32 zero = 0;
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/http2/decoding.h:284]:
```suggestion
    const __u32 zero = 0;
```

File: pkg/network/protocols/http2/protocol.go [hunk 2]
```diff
@@ -129,7 +129,7 @@ func (p *protocol) Name() string {
 }
 
 const (
-	mapSizeValue = 1024
+	mapSizeValue = 10240
```
Comment (amitslavin) [inline pkg/network/protocols/http2/protocol.go:132]: will be changed as part of this [PR](https://github.com/DataDog/datadog-agent/pull/20955/files)

**Review by amitslavin — CHANGES_REQUESTED** [review]

**Comment by amitslavin** [comment]
@guyarb Could you please provide some details regarding the outcomes of the modification? Specifically, I'm interested in knowing the size of the map before and after the change

File: pkg/network/protocols/http2/protocol.go (hunk 2 above)
Comment (guyarb) [reply pkg/network/protocols/http2/protocol.go:132]: good catch!

**Review by amitslavin — APPROVED** [review]

**Pull request merged**